DeviceAdapters/MicroPoint/MicroPoint.cpp

```diff
@@ -185,7 +185,7 @@ double MicroPoint::AttenuatorTransmissionFromIndex(long n)
 
 int MicroPoint::StepAttenuatorPosition(bool positive)
 {
-   unsigned char buf[] = {'C', (positive ? 0xc0 : 0x80), 'C', 0x00};
+   unsigned char buf[] = {'C', (unsigned char)(positive ? 0xc0 : 0x80), 'C', 0x00};

```

Comment:
Just to note for the record, these casts to `unsigned char` are probably needed to compile in C++11 (or later) mode -- they should not be needed if `-std=c++03` is added to the compiler flags. But we'd like to move to modern C++ soon, so thanks!